src/python/pants/engine/isolated_process.py

```diff
@@ -6,36 +6,44 @@
                         unicode_literals, with_statement)
 
 import logging
+import six
 
 from pants.engine.fs import EMPTY_SNAPSHOT, DirectoryDigest
 from pants.engine.rules import RootRule
-from pants.util.objects import TypeCheckError, datatype
+from pants.util.objects import Exactly, SubclassesOf, TypeCheckError, datatype
 
 
 logger = logging.getLogger(__name__)
 
+_default_timeout_seconds = 15 * 60
+
 
 class ExecuteProcessRequest(datatype([
   ('argv', tuple),
   ('env', tuple),
   ('input_files', DirectoryDigest),
-  ('output_files', tuple)
+  ('output_files', tuple),
+  # NB: timeout_seconds covers the whole remote operation including queuing and setup.
+  ('timeout_seconds', Exactly(float, int)),
+  ('description', SubclassesOf(*six.string_types)),
 ])):
   """Request for execution with args and snapshots to extract."""
 
   @classmethod
-  def create_from_snapshot(cls, argv, env, snapshot, output_files):
+  def create_from_snapshot(cls, argv, env, snapshot, output_files, timeout_seconds=_default_timeout_seconds, description='process'):
     cls._verify_env_is_dict(env)
     return ExecuteProcessRequest(
       argv=argv,
       env=tuple(env.items()),
       input_files=snapshot.directory_digest,
       output_files=output_files,
+      timeout_seconds=timeout_seconds,
+      description=description,
     )
 
   @classmethod
-  def create_with_empty_snapshot(cls, argv, env, output_files):
-    return cls.create_from_snapshot(argv, env, EMPTY_SNAPSHOT, output_files)
+  def create_with_empty_snapshot(cls, argv, env, output_files, timeout_seconds=_default_timeout_seconds, description='process'):
```

Comment:
I believe that this could be merged with the previous method by giving the `snapshot` a default argument of `snapshot=DEFAULT_SNAPSHOT`.